### ISSUE TITLE
Auth facade doesn't recognize macros registered on SessionGuard,RequestGuard

### ISSUE THREAD

**Issue by dianfishekqi** [nunomaduro/larastan#1365]
- Larastan Version: 9.27.0
- `--level` used: 8

### Description

```
Call to an undefined method Illuminate\Support\Facades\Auth::authed().  
```

### Laravel code where the issue was found

```php
$authedMacro = function (): Model {
    if (($auth = Auth::user()) !== null) {
        return $auth;
    }

    throw new UnauthorizedException();
};

SessionGuard::macro('authed', $authedMacro);
RequestGuard::macro('authed', $authedMacro);
```

**Comment by szepeviktor** [comment]
Thank you for your report!

What happens if you temporarily try this.
```php
SessionGuard::macro('authed', function (): Model {
    if (($auth = Auth::user()) !== null) {
        return $auth;
    }

    throw new UnauthorizedException();
});
```

**Comment by dianfishekqi** [comment]
Hello there, 

I get the same error.

**Comment by szepeviktor** [comment]
Could you share the code where this error happens?

**Comment by dianfishekqi** [comment]
Inside `OrderController`
```php
$user = Auth::authed()
```

**Comment by xwillq** [comment]
If I understood it correctly, this bug is caused by #1295. I proposed solutions to this in #1315. Second solution, rollback the PR and fix that issue in some other way, will fix this issue as well. After rollback the logic should be something like this:
1. Pipes\Facades will get underlying class of facade Auth and send it back to pipeline
2. Pipes\Auths will send Guard contract to pipeline
3. Pipes\Contracts will resolve Guard contract to default guard
4. Pipes\Macros will find macro function inside default guard

Right now step 4 is missing because check for macros was moved to separate extension.
